test/unit/multiplication/test_multiplication_triangular.cpp

```diff
@@ -57,11 +68,23 @@ GlobalElementSize globalTestSize(const LocalElementSize& size) {
   return {size.rows(), size.cols()};
 }
 
+template <class T>
+
+std::tuple<std::function<T(const GlobalElementIndex&)>, std::function<T(const GlobalElementIndex&)>,
+           std::function<T(const GlobalElementIndex&)>>
+getTriangularSystem(blas::Side side, blas::Uplo uplo, blas::Op op, blas::Diag diag, T alpha, SizeType m,
+                    SizeType n) {
+  if (side == blas::Side::Left)
+    return getLeftTriangularSystem<GlobalElementIndex, T>(uplo, op, diag, static_cast<T>(1.0) / alpha,
+                                                          m);
+  else
+    return getRightTriangularSystem<GlobalElementIndex, T>(uplo, op, diag, static_cast<T>(1.0) / alpha,
+                                                           n);
+}
```

Comment:
I would put it in `` such that it can be reused anywhere it is needed.
Anyway I would write it in the following way:
```suggestion
template <class ElementIndex, class T, class ...Args>
auto getTriangularSystem(blas::Side side, Args ...args) {
  if (side == blas::Side::Left)
    return getLeftTriangularSystem<ElementIndex, T>(args...);
  else
    return getRightTriangularSystem<ElementIndex, T>(args...);
}
```

Reply:
Note: `1. / alpha` remains on the caller side.